mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RegistrationQueueName.java

```diff
@@ -0,0 +1,47 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+
+public class RegistrationQueueName {
+    private Optional<String> queueName;
+
+    RegistrationQueueName() {
+        this.queueName = Optional.empty();
+    }
+
+    void initialize(String queueName) {
+        Preconditions.checkNotNull(queueName);
+        Preconditions.checkState(!this.queueName.isPresent(), "");
+        this.queueName = Optional.of(queueName);
+    }
+
+    String asString() {
+        Preconditions.checkState(queueName.isPresent(), "'queueName' is not yet initialized");
+        return queueName.get();
```

Comment:
This can be written:

```
    String asString() {
        return queueName.orElseThrow(() -> new IllegalStateException("'queueName' is not yet initialized"));
    }
```

Calling isPresent then get is an Optional misuse and anti-pattern.